ceremonies/src/lib.rs

```diff
@@ -944,6 +951,45 @@ impl<T: Config> Pallet<T> {
 		Ok(participant_type)
 	}
 
+	/// removes a participant from the registry maintaining continuous indices
+	fn unregister_newbie(
+		cid: CommunityIdentifier,
+		cindex: CeremonyIndexType,
+		participant: &T::AccountId,
+	) -> Result<(), Error<T>> {
+		if <encointer_scheduler::Pallet<T>>::current_phase() != CeremonyPhaseType::REGISTERING {
+			return Err(<Error<T>>::WrongPhaseForUnregistering.into())
+		}
+
+		if !<NewbieIndex<T>>::contains_key((cid, cindex), &participant) {
+			return Ok(())
+		}
+
+		let participant_index = <NewbieIndex<T>>::get((cid, cindex), &participant);
+		let participant_count = <NewbieCount<T>>::get((cid, cindex));
+
+		<NewbieRegistry<T>>::remove((cid, cindex), &participant_index);
+		<NewbieIndex<T>>::remove((cid, cindex), &participant);
+
+		for idx in (participant_index + 1)..=participant_count {
+			let p = <NewbieRegistry<T>>::get((cid, cindex), &idx)
+				.expect("idx <= participant_count, so p is registered; qed");
+
+			let new_idx = idx - 1;
+			<NewbieIndex<T>>::insert((cid, cindex), &p, new_idx);
+			<NewbieRegistry<T>>::remove((cid, cindex), &idx);
+			<NewbieRegistry<T>>::insert((cid, cindex), &new_idx, &p);
+		}
+		<NewbieCount<T>>::insert(
+			(cid, cindex),
+			participant_count
+				.checked_sub(1)
+				.expect("participant is registered, thus participant_count > 0; qed"),
+		);
```

Comment:
Why this for-loop? Is there a reason preventing us replacing the missing index with the participant with the highest index? This would be way more efficient.

Reply:
You could use the built-in storage-map method [swap](https://github.com/paritytech/substrate/blob/26eff96fed52fb2001f37f17e743939b0e2384bb/frame/support/src/storage/generator/double_map.rs#L165) to first swap the index with the last entry and then remove the last entry.

Reply:
this would indeed be a lot more efficient. but it would not keep the order of registration which would be unfair in the case where not all newbies get assigned to a meetup.

Reply:
but we could argue that nobody gets a disadvantage, just one participant gets an advantage and this could be acceptable for the big performance gain...

Reply:
a for loop is inacceptable anyway as it is O(n) where n is the number of newbies which is unbounded

Reply:
can the system be played? is it predictable who gets the advantage from endorsing a newbie a posteriori?

Reply:
yes if there are `n` newbies with indices `1..=n`, and newbie with index `i` gets endorsed, then newbie with index `n` will receive index `i`

Reply:
but as a regular user in order to play the system, you would have to know exactly when the endorsement will happen

Reply:
I think it's acceptable. not nice, but much better than O(n)